### PR TITLE
fix: changes the way empty vars are handled on deploy-admission-contr…

### PR DESCRIPTION
**Pull request by thgruiz**
**What this PR does / why we need it**:

When trying to install the admission webhook (following here: https://docs.konghq.com/kubernetes-ingress-controller/2.8.x/deployment/admission-webhook/) because of some empty vars like KUBECONFIG_OPTION and others with wrong double quotes (https://www.shellcheck.net/wiki/SC2086 see Exceptions) I found the issues bellow on an ubuntu 22.04:

1 - on first kubectl command:
    error: exactly one NAME is required, got 2

2 - when executing the patch on ingress-kong deployment:
    error: there is no need to specify a resource type as a separate argument when passing arguments in resource/name form (e.g. 'kubectl get resource/<resource_name>' instead of 'kubectl get resource resource/<resource_name>'

3 - the base64 command gives me the error:
    base64: '"-w0"': No such file or directory

**Which issue this PR fixes**:

I looked, but could not found open issues for this.

**Special notes for your reviewer**:

Just tested on my ubuntu, with both server and client on 1.22 and 1.26. Don't know where you guys use this script, so...

### CODE REVIEW

**Review by rainest — COMMENTED** [review]
Is there a way to fix this without the `export`? That will affect the current shell's kubectl configuration past the script, correct?

**Comment by pmalek** [comment]
@thgruiz Thanks for the patch!

As I look at it, it seems that we don't really use this script. It's only referred from 1 place: https://github.com/Kong/kubernetes-ingress-controller/blob/c63fbca0ec3782f5a3b2a9400a19474d771d0c8d/test/e2e/features_test.go#L72 which is unused.

I wonder if it wouldn't be better to remove this altogether (together with `admissionScriptPath` const) rather than trying to keep it alive. @rainest WDYT?

**Comment by thgruiz** [comment]
> export

That will affect the shell only if you source the script somewhere, but since there is a #!/bin/bash I don't think this script is meant to be sourced...

**Comment by thgruiz** [comment]
> Is there a way to fix this without the `export`? That will affect the current shell's kubectl configuration past the script, correct?


export

That will affect the shell only if you source the script somewhere, but since there is a #!/bin/bash I don't think this script is meant to be sourced...

**Review by rainest — DISMISSED** [review]
Ah, okay. I don't think in shell scope often, that makes sense now after running through it.

@pmalek I don't think we reference it within the repo, but I think we might in (some) docs. We can possibly move it entirely to the docs site, but the fix is the same regardless.

**Comment by thgruiz** [comment]
I think something is missing in the ci configuration, since we got this on the logs:

Error: tests failed: no license could be found because KONG_LICENSE_DATA was not set

**Comment by pmalek** [comment]
> I think something is missing in the ci configuration, since we got this on the logs:
> 
> Error: tests failed: no license could be found because KONG_LICENSE_DATA was not set
> 
> Sorry, I don't know I can fix this alone...

Please rebase on recent `main`. The recent changes in CI should make this pass for you. The failure is caused by our workflows not providing secrets like Kong license to PRs from forks.

**Comment by thgruiz** [comment]
done @pmalek, sorry for the delay

had to force push my rebase since the 'Sync fork' on github web interface did a merge (first time that I used that sync...)
now, after ci runs, I think its good to merge

File: hack/deploy-admission-controller.sh [hunk 1]
```diff
@@ -73,4 +74,6 @@ webhooks:
     service:
       namespace: kong
       name: kong-validation-webhook
-    caBundle: $(cat ${TMPDIR}/tls.crt | base64 \"${BASE64_OPTIONS}\")" | kubectl apply "${KUBECONFIG_OPTION}" -f -
+    caBundle: $(base64 ${BASE64_OPTIONS:+${BASE64_OPTIONS}} "${TMPDIR}/tls.crt") 
+EOF
+) | kubectl apply -f -
```
Comment (pmalek) [inline hack/deploy-admission-controller.sh:79]:
```suggestion
) | kubectl apply -f -

```

**Review by pmalek — APPROVED** [review]

**Comment by pmalek** [comment]
@thgruiz Thanks for the contribution!

**Pull request merged**